Tensile/SolutionStructs.py

```diff
@@ -2454,10 +2455,20 @@ def assignDerivedParameters(state):
     #These modes only work under certain conditions, apply them here:
     #  - The "NoLoad" loop is only generated if PrefetchGlobalRead>0
     #  - And Suppress does not work if GSU>1 for some reason
-    state["SuppressNoLoadLoop"] &= (bufferLoad and state["PrefetchGlobalRead"] and (state["GlobalSplitU"]==1))
-    # Pointer swap only used if PGR=1 - so set ExpandPointerSwap=0 here
-    # EPS not supported with PAP yet
-    state["ExpandPointerSwap"]  &= (bufferLoad and state["PrefetchGlobalRead"] and not state["PrefetchAcrossPersistent"])
+    if state["SuppressNoLoadLoop"] == 1:
+      if not (bufferLoad and state["PrefetchGlobalRead"] and (state["GlobalSplitU"]==1)):
+        state["SuppressNoLoadLoop"] = 0
+
+    if state["ExpandPointerSwap"] == 1:
+      # Pointer swap only used if PGR=1 - so set ExpandPointerSwap=0 here
+      if not (bufferLoad and state["PrefetchGlobalRead"] == 1):
+        state["ExpandPointerSwap"] = 0
+      # EPS not supported with PGR=2 yet
+      if state["PrefetchGlobalRead"] == 2:
+        state["ExpandPointerSwap"] = 0
+      # EPS not supported with PAP yet
+      if state["PrefetchAcrossPersistent"]:
+        state["ExpandPointerSwap"] = 0
```

Comment:
Should these param modifications have a printed warning, or be rejected instead of adjusted?